File: Docs/WSC18/data_structure_eval/simian-maths/cheap_ctype/test_heap.py
import heap
from random import randint
import sys
import timeit

REPEATS = 1
NUMBER = 1

# push in increasing

def test1():
    pq = heap.init()
    for x in range(1000000) :
        heap.push(pq,(x,
                      {"rx" : "a",
                       "tx" : "b",
                       "txID" : 1,
                       "rxID" : 1,
                       "name" : "bob",         
                       "data": "none",
                      }))
        #print x
    #for x in range(1000000) :
    #    y=heap.peak(pq)
    
    for x in range(1000000):
        #y = heap.pop(pq)
        #print y[0] #print y
        #print "\n"
        pass

def test2():
    pq = heap.init()

    for x in range(1000000) :
        
        heap.push(pq,(randint(0,10000),
                      {"rx" : "a",
                       "tx" : "b",
                       "txID" : 1,
                       "rxID" : 1,
                       "name" : "bob",         
                       "data": "none",
                      }))
        
    #for x in range(10000) :
    #    y=heap.peak(pq)
        
    #for x in range(1000000):
    #    y = heap.pop(pq)
            #print y          


def test3():
    pq = heap.init()
    totalPushes=0
    pushes=0
    while(totalPushes<1000000):
        for x in range(randint(1,100)):
            heap.push(pq,(randint(0,10000),
                          {"rx" : "a",
                           "tx" : "b",
                           "txID" : 1,
                           "rxID" : 1,
                           "name" : "bob",         
                           "data": "none",
                          }))
            totalPushes+=1
            pushes+=1
        #for x in range(10000) :
        #    y=heap.peak(pq)
        
        for x in range(randint(1,100)):
            if (pushes):
                pushes-=1
                #y = heap.pop(pq)
            else:
                break
            #print y          



#if test == 1:
print min(timeit.Timer(test1).repeat(repeat=REPEATS, number=NUMBER))/NUMBER     
#if test ==2:
print min(timeit.Timer(test2).repeat(repeat=REPEATS, number=NUMBER))/NUMBER     
print min(timeit.Timer(test3).repeat(repeat=REPEATS, number=NUMBER))/NUMBER     

#test1()

File: Docs/WSC18/data_structure_eval/simian-datastructs/lua/test_heap.lua
jit.opt.start(3, 'hotloop=3', 'hotexit=8', 'instunroll=10', 'loopunroll=10', 'callunroll=10', 'recunroll=10', 'tryside=30') --Optional JIT optimization flags: user adjustable           

local eventQ = require "eventQ"

local eventQueue = {}

eventQ.push(eventQueue, event)

event = eventQ.pop(eventQueue)




-- test 1
for NUMBER=1,10 do

   for x=1,1000000 do
      local event = {
	 tx = 'a',
	 rx = 'b',
	 txID = 1,
	 rxID = 1,
	 name = "bob",
	 data = "none",
	 time = x,
      }
      eventQ.push(eventQueue,event)
   end
   
   for y=1,1000000 do
      event = eventQ.pop(eventQueue)
   end
   
end


--[[
-- test 2
local x,y
local NUMBER
local mat = math.random
for NUMBER=1,10 do

   for x=1,1000000 do
      local tt = mat(1000000)
      local event = {
	 tx = 'a',
	 rx = 'b',
	 txID = 1,
	 rxID = 1,
	 name = "bob",
	 data = "none",
	 time = tt,
      }
      eventQ.push(eventQueue,event)
   end
   
   for y=1,1000000 do
      event = eventQ.pop(eventQueue)
   end
   
end
]]

-- test 3
--[[
local x,y
local totalPushes = 0
local pushes = 0
local NUMBER
local mat = math.random
for NUMBER=1,10 do
   totalPushes = 0
   pushes = 0
   while totalPushes < 1000000 do
      for x=1,mat(100) do --1000000 do
	 pushes = pushes + 1
	 totalPushes = totalPushes + 1
	 local tt = mat(1000000)
	 local event = {
	    tx = 'a',
	    rx = 'b',
	    txID = 1,
	    rxID = 1,
	    name = "bob",
	    data = "none",
	    time = tt,
	 }
	 eventQ.push(eventQueue,event)
      end
      
      for y=1,mat(100) do --1000000 do
	 if pushes <= 0 then
	    break
	 end
	 pushes = pushes - 1
	 event = eventQ.pop(eventQueue)
	 
	 
      end
   end

end
]]
